ingestion/src/metadata/ingestion/source/database/snowflake/metadata.py

```diff
@@ -457,7 +500,182 @@ def yield_life_cycle_data(self, _) -> Iterable[Either[OMetaLifeCycleData]]:
                     )
                 )
         except Exception as exc:
-            logger.debug(traceback.format_exc())
-            logger.error(
-                f"Unable to get the table life cycle data for table {table.name.__root__}: {exc}"
+            yield Either(
+                left=StackTraceError(
+                    name=table.name.__root__,
+                    error=f"Unable to get the table life cycle data for table {table.name.__root__}: {exc}",
+                    stack_trace=traceback.format_exc(),
+                )
             )
+
+    def get_stored_procedures(self) -> Iterable[SnowflakeStoredProcedure]:
+        """List Snowflake stored procedures"""
+        if self.source_config.includeStoredProcedures:
+            results = self.engine.execute(
+                SNOWFLAKE_GET_STORED_PROCEDURES.format(
+                    database_name=self.context.database.name.__root__,
+                    schema_name=self.context.database_schema.name.__root__,
+                )
+            ).all()
+            for row in results:
+                stored_procedure = SnowflakeStoredProcedure.parse_obj(dict(row))
+                yield stored_procedure
+
+    def yield_stored_procedure(
+        self, stored_procedure: SnowflakeStoredProcedure
+    ) -> Iterable[Either[CreateStoredProcedureRequest]]:
+        """Prepare the stored procedure payload"""
+
+        try:
+            yield Either(
+                right=CreateStoredProcedureRequest(
+                    name=EntityName(__root__=stored_procedure.name),
+                    description=stored_procedure.comment,
+                    storedProcedureCode=StoredProcedureCode(
+                        language=STORED_PROC_LANGUAGE_MAP.get(
+                            stored_procedure.language
+                        ),
+                        code=stored_procedure.definition,
+                    ),
+                    databaseSchema=self.context.database_schema.fullyQualifiedName,
+                    sourceUrl=SourceUrl(
+                        __root__=f"https://app.snowflake.com/{self.region.lower()}"
+                        f"/{self.account.lower()}/#/data/databases/{self.context.database.name.__root__}"
+                        f"/schema/{self.context.database_schema.name.__root__}"
```

Comment:
most of this source url is already build in the `get_source_url` method
Might be better to reuse that method by adding an if else for table and procedure and keep the logic centralised